WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -350,13 +354,12 @@ + (NSArray *)routes
   return FBResponseWithOK();
 }
 
-+ (XCUICoordinate *)getGestureCoordinate:(FBRouteRequest *)request
++ (XCUICoordinate *)gestureCoordinateWithCoordinate:(CGPoint)coordinate application:(XCUIApplication *)application
 {
-  FBSession *session = request.session;
-  CGVector point = CGVectorMake((CGFloat)[request.arguments[@"x"] doubleValue], (CGFloat)[request.arguments[@"y"] doubleValue]);
-  XCUICoordinate *appCoordinate = [[XCUICoordinate alloc] initWithElement:session.application normalizedOffset:CGVectorMake(0, 0)];
-  XCUICoordinate *gestureCoordinate = [[XCUICoordinate alloc] initWithCoordinate:appCoordinate pointsOffset:point];
-  return gestureCoordinate;
+  // TODO: handle this properly after XCTest starts to respect landscape interface orientation
+  CGPoint point = FBInvertPointForApplication(coordinate, application.frame.size, application.interfaceOrientation);
```

Comment:
Yes, but all versions checks etc, would happen in that class. So final product would be "transparent" for all clients. All of them should expect correct values anyway. Only problem would be argument you made in first point.